api/routes/api/auth.js

```diff
@@ -8,56 +8,89 @@ require('dotenv').config();
 
 const User = require('../../models/User');
 
-// Get authorized user
-router.get('/', auth, async (req, res) => {
-  try {
-    const user = await User.findById(req.user.id).select('-password');
-    res.json(user);
-  } catch (err) {
-    console.error(err.message);
-    res.status(500).send('Server error');
-  }
-});
+// Register user
+router.post('/register', async (req, res) => {
+
+    const { username, email, password } = req.body;
 
-// Authenticate user & get token
-router.post(
-  '/',
-  [
-    check('email', 'Email is required').isEmail(),
-    check('password', 'Password is required').exists(),
-  ],
-  async (req, res) => {
-    const errors = validationResult(req);
-    if (!errors.isEmpty()) {
-      return res.status(400).json({ errors: errors.array() });
+    const emailRegexp = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/
+  
+    const validation = {
+      email: {test: emailRegexp.test(email), error:'invalid email format'}, 
+      username: {test: username?.length>=3, error: 'username must be at least 3 characters'}, 
+      password: {test: password.length>=6, error:'password must be at least 6 characters'}
+    };
+    
+    const missingFields = [];
+    const emptyFields = [];
```

Comment:
missing and empty fields should be handled on the frontend, a request with missing fields shouldn't go through.